pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -1279,10 +1280,11 @@ func (qjm *XController) ScheduleNext() {
 					}
 				} else {
 					klog.V(10).Infof("[ScheduleNext] HOL quota evaluation not enabled for %s at %s activeQ=%t Unsched=%t &qj=%p Version=%s Status=%+v due to quota limits", qj.Name, time.Now().Sub(HOLStartTime), qjm.qjqueue.IfExistActiveQ(qj), qjm.qjqueue.IfExistUnschedulableQ(qj), qj, qj.ResourceVersion, qj.Status)
+					awWithoutQuota = true
 				}
 
 				// If quota evalauation is set or quota evaluation not enabled set the appwrapper to be dispatched
-				if fits {
+				if fits || awWithoutQuota {
```

Comment:
Can this just just be changed to:
```suggestion
				if fits || !qjm.serverOption.QuotaEnabled {
```
and remove the new variable?

Reply:
I believe the intention of the new variable is different than what `qjm.serverOption.QuotaEnable` is doing. The issue is to clean the quota tree if the AppWrapper was deleted but the quota was not released correctly. Right @asm582 ?

Reply:
This was pushed by mistake